server/etcdserver/txn/txn_test.go

```diff
@@ -367,6 +370,26 @@ func TestWriteTxnPanic(t *testing.T) {
 		},
 	}
 
+	// compute DB file hash before applying the txn
+	dbHashBefore, err := computeFileHash(bePath)
+	if err != nil {
+		t.Fatalf("failed to compute DB file hash before txn: %v", err)
+	}
+
+	// we verify the following things below:
+	// - server panics after a write txn aply fails (invariant being tested: server should never try to move on from a failed write)
+	// - no writes from the txn are applied to the backend (invariant being tested: failed write should have no side-effect on DB state besides panic)
+	defer func() {
+		dbHashAfter, err := computeFileHash(bePath)
+		if err != nil {
+			t.Fatalf("failed to compute DB file hash after txn: %v", err)
+		}
+
+		if dbHashAfter != dbHashBefore {
+			t.Fatalf("expected DB hash after failed txn: %s, got: %s", dbHashBefore, dbHashAfter)
+		}
```

Comment:
What's the recommendation here? `require.Equalf` ? cc @mmorel-35

Reply:
thanks for this suggestion - made it much cleaner